src/components/HdExpandText.vue

```diff
@@ -33,28 +33,18 @@
 <script>
 import merge from 'lodash/merge';
 import { getMessages } from 'hd-blocks/lang';
-import { isMobile } from 'hd-blocks/services/breakpoints';
+import { mediaMatches } from 'hd-blocks/services/breakpoints';
 import onResize from 'hd-blocks/services/on-resize';
 
-const getApproximatedLineHeight = () => {
-  if (isMobile()) {
-    return 28;
-  }
-
-  return 32;
-};
-
 export default {
   name: 'HdExpandText',
   props: {
+    // Can be an object with keys set to breakpoints and values the desired
+    // lines count
     lines: {
-      type: Number,
+      type: [Number, Object],
```

Comment:
I'm personally against multiple types. I would have 2 different properties and for the "linesBreakpoints" I would also add the validator.

Reply:
Why so? If we have two properties that basically do the same thing, how do we determine which one takes precedence?

I can add a validator for the argument, that might be a good idea (since it's a form of documentation on its own).

Reply:
I don't like it because you've to do `typeof` checks in the code to guarantee different behaviors. But as I said, it is personal only.

Reply:
Yeah, I can understand that. I don't think it's ideal either. However, I don't see a version with two different props as any better. The `typeof` checks would have to be replaced with checking whether one argument is set, and if not falling back to another (in essence doing the same thing, differently), but with a drawback of having an additional prop to maintain and arbitrarily deciding which one takes preference.

Hopefully, the rest of the team weighs in and we can adjust as per majority preference.

Reply:
I checked how some other component libraries are doing it and some of them are also following the "multiple types" approach (for example `Button` component in Vuetify for the `href` prop supports a `string` or an `object`, `Dropdown` component inBuefy for the `can-close` prop supports `boolean` and `array`.

Not saying that proves it's the best way to do it, but we would not be setting a precedent :smile:

Reply:
Hahahah, great! Nice catch! I'm already convinced about multiple types hehehehe

Reply:
I have no strong preference here, both have ups and downs, as you've discussed. ;)